deploy/04_deploy_vending_machine.ts

```diff
@@ -0,0 +1,119 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
```

Comment:
The solution proposed in this script seems very sophisticated. I'm afraid it might be too complicated. What if we go with a much simpler script, consistent with the other ones, and just enhance the previous version https://github.com/threshold-network/solidity-contracts/blob/32d5689e1e3c101c74f09102a5c5173d6aac7969/deploy/03_deploy_vending_machine.ts8 by adding a few lines with a deployment of another `VendingMachine` for NU:
```ts
 const T_ALLOCATION_NU = tTotalSupply.mul(45).div(100)

  const vendingMachine = await deployments.deploy("VendingMachineNuCypher", {
    contract: "VendingMachine",
    from: deployer,
    args: [NuCypherToken.address, T.address, nuTotalSupply, T_ALLOCATION_NU],
    log: true,
  })
```

IMHO the deployment scripts should be kept simple and easily maintainable. There will be just two types of vending machines, so we don't have to think about being too universal.

Reply:
Yeah, I see your point. Should we split scripts into two files eg.`04_deploy_vending_machine_keep` and `05_deploy_vending_machine_nu`? What do you think?

Reply:
I don't have a strong opinion about splitting the script. I can see the advantages of both solutions. Up to you!

Reply:
Please see cab0c53 and 03416c4.